```
    And Run "firewall-cmd --zone=public --add-rich-rule='rule port port=22 protocol=tcp drop' --permanent && firewall-cmd --reload" on "hanode2" # opt/crmsh/test/features/steps/step_implementation.py:109
      Traceback (most recent call last):
        File "/usr/lib/python3.13/site-packages/behave/model.py", line 1991, in run
          match.run(runner.context)
          ~~~~~~~~~^^^^^^^^^^^^^^^^
        File "/usr/lib/python3.13/site-packages/behave/matchers.py", line 105, in run
          self.func(context, *args, **kwargs)
          ~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "opt/crmsh/test/features/steps/step_implementation.py", line 111, in step_impl
          _, out, _ = run_command_local_or_remote(context, cmd, addr)
                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^
        File "/opt/crmsh/test/features/steps/utils.py", line 100, in run_command_local_or_remote
          results = list(executor.map(lambda x: (x, behave_agent.call(x, 1122, cmd, user=user)), hosts))
        File "/usr/lib64/python3.13/concurrent/futures/_base.py", line 619, in result_iterator
          yield _result_or_cancel(fs.pop())
                ~~~~~~~~~~~~~~~~~^^^^^^^^^^
        File "/usr/lib64/python3.13/concurrent/futures/_base.py", line 317, in _result_or_cancel
          return fut.result(timeout)
                 ~~~~~~~~~~^^^^^^^^^
        File "/usr/lib64/python3.13/concurrent/futures/_base.py", line 456, in result
          return self.__get_result()
                 ~~~~~~~~~~~~~~~~~^^
        File "/usr/lib64/python3.13/concurrent/futures/_base.py", line 401, in __get_result
          raise self._exception
        File "/usr/lib64/python3.13/concurrent/futures/thread.py", line 59, in run
          result = self.fn(*self.args, **self.kwargs)
        File "/opt/crmsh/test/features/steps/utils.py", line 100, in <lambda>
          results = list(executor.map(lambda x: (x, behave_agent.call(x, 1122, cmd, user=user)), hosts))
                                                    ~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/opt/crmsh/test/features/steps/behave_agent.py", line 69, in call
          s.connect(sockaddr)
          ~~~~~~~~~^^^^^^^^^^
      PermissionError: [Errno 13] Permission denied
```

--- a/test/features/bootstrap_firewalld.feature
+++ b/test/features/bootstrap_firewalld.feature
@@ -37,8 +37,6 @@ Feature: Test deployment of high-availability firewalld service
     And     The "high-availability" firewalld service is not added on "hanode1"
     And     The "high-availability" firewalld service is not added on "hanode2"
     # open behave agent port
-    When    Run "firewall-offline-cmd --add-port=1122/tcp" on "hanode1"
-    When    Run "firewall-offline-cmd --add-port=1122/tcp" on "hanode2"
     When    Run "systemctl start firewalld" on "hanode2"
     When    Run "systemctl start firewalld" on "hanode1"
     Then    Service "firewalld" is "started" on "hanode2"
@@ -62,7 +60,6 @@ Feature: Test deployment of high-availability firewalld service
     Given   Cluster service is "started" on "hanode1"
     And     Cluster service is "started" on "hanode2"
     And     Service "firewalld" is "stopped" on "qnetd-node"
-    When    Run "firewall-offline-cmd --add-port=1122/tcp" on "qnetd-node"
     When    Run "systemctl start firewalld" on "qnetd-node"
     Then    Service "firewalld" is "started" on "qnetd-node"
     When    Run "crm cluster init qdevice --qnetd-hostname=qnetd-node -y" on "hanode1"

--- a/test/run-functional-tests
+++ b/test/run-functional-tests
@@ -213,6 +213,7 @@ deploy_ha_node() {
 	podman_exec $node_name "echo 'StrictHostKeyChecking no' >> /etc/ssh/ssh_config"
 	podman_exec $node_name "systemctl start sshd.service"
 	podman_exec $node_name "systemctl stop firewalld.service"
+    podman_exec $node_name "firewall-offline-cmd --add-port=1122/tcp"
 
 	if [[ "$node_name" != "qnetd-node" && ! "$node_name" =~ ^pcmk-remote-node[0-9]$ ]];then
 	  podman cp $PROJECT_PATH $node_name:/opt/crmsh

